src/load.ts

```diff
@@ -54,16 +54,16 @@ export function resolveSchema(url: string): URL {
 interface LoadOptions extends GlobalContext {
   rootURL: URL;
   schemas: SchemaMap;
+  urlCache?: Set<string>; // URL cache (prevent URLs from being loaded over and over)
 }
 
-// temporary cache for load()
-let urlCache = new Set<string>(); // URL cache (prevent URLs from being loaded over and over)
-
 /** Load a schema from local path or remote URL */
 export default async function load(
   schema: URL | PartialSchema,
   options: LoadOptions
 ): Promise<{ [url: string]: PartialSchema }> {
+  const urlCache = options.urlCache || new Set<string>();
```

Comment:
How can we ensure that the cache object being passed in is of type `Set` when the user is consuming the library with vanilla JS? They will have types provided, but it will not throw an error if they pass a different data structure.

Reply:
Thanks for asking. As I see this function is not a part of the library public api. And the function uses the cache object for recursive calls optimization. Also as you may see the function does not check types of the other fields of LoadOptions.